### ISSUE TITLE
Buildcraft 7+ texture

### ISSUE THREAD

**Issue by Unrepentant-Atheist** [F32Organization/Faithful32-1.7.10#331]
Textures for Buildcraft Version 7+ need to be redistributed as Buildcraft is made modular. The names are the same, just not 1 folder anymore.

**Comment by Evolution0** [comment]
This is problematic as inside the jars the textures are not in an assets folder, they are just loose files mixed in with the class files which would mean that all BC7 textures would need to be present in the root of the repo.

Now they *can* be put in a folder for the sake of keeping things neat and later moved by the script during packaging.

Might want to wait for a second opinion on this though the modifications are super simple.

**Comment by Unrepentant-Atheist** [comment]
What do ya mean problematic? All ya gotta do is create folders and redistribute the textures y'all currently got in buildcraft. The new folder structure is this one: http://i.imgur.com/ebjOGTr.png

**Comment by Evolution0** [comment]
I know *that*, I couldn't figure out the folder structure though as when you pop open the jar files there are no folders inside. I had to go over to the buildcraft repo and take a peek at the source just to figure it out.

So by problematic I was under the assumption that they weren't even using an assets directory which would have made a mess.

**Comment by Unrepentant-Atheist** [comment]
You do know, you can rename *.jar into *.zip and open it up?

**Comment by StefanTT** [comment]
Just as a quick side node: I would prefer to not drop BC6 support yet, as most mod packs still use BC6.

**Comment by Evolution0** [comment]
I've already mentioned twice that I *have* opened them up. There are no folders but there *are* loose class and image files.

That is irrelevant now though, I've created the modular folder structure based on the BC repo and moved over what was salvageable, I've left the existing BC folder untouched to maintain BC6 support.

If a user adds the BC7 (non-modular) jar does it use the main BC folder to the separate folders?

Basically what I'm asking is are the modular texture folders only used when using the modular jar files?

**Comment by StefanTT** [comment]
Urks that looks very ugly (having the images in the root folder)

**Comment by StefanTT** [comment]
Ok I had a look at the JARs, just to be sure. Downloaded both the all-in-one and some modular JARs of bc 7.0.8 from http://mod-buildcraft.com/pages/download.html

For me they are properly packaged with assets/*/textures folders.

**Comment by Evolution0** [comment]
Then its most likely that peazip is collapsing everything in flat mode and showing me only the files, disabling flat mode hides any directories and shows me just the licence.

No big deal, I can check the full name on the right it seems which has the full path in it.

**Comment by GamerM** [comment]
As of right now i'm going to work on the redistribution of the files. As of buildcraft 7.0.13 all texture seen before in the Buildcraft folder assets/buildcraft are completely remove. Does that mean everything inside the folder need to be cut and pasted to the right folder. I'm seeing double copies tbh.
